lib/at_cmd_parser/at_utils.h

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file at_utils.h
+ *
+ * @defgroup at_cmd_parser_utils AT Command utility functions.
+ * @ingroup at_cmd_parser
+ * @{
+ * @brief AT parser utility functions to deal with strings.
+ */
+#ifndef AT_UTILS_H__
+#define AT_UTILS_H__
+
+#include <zephyr/types.h>
+#include <stddef.h>
+
+#define AT_PARAM_SEPARATOR              ','
+#define AT_CMD_SEPARATOR                ':'
+#define AT_CMD_BUFFER_TERMINATOR        0
+#define AT_CMD_STRING_IDENTIFIER        '\"'
+#define AT_STANDARD_NOTIFICATION_PREFIX '+'
+#define AT_PROP_NOTIFICATION_PREFX      '%'
+
+/**
+ * @brief Check if character is a notification start character
+ *
+ * This function will check if the character is a + or % which identifies
+ * a notification from the AT host.
+ *
+ * @param[in] char Character that should be examined
+ *
+ * @retval true  If character is + or %
+ * @retval false If character is something else
+ */
+static inline bool is_notification(char chr)
+{
+	if ((chr == AT_STANDARD_NOTIFICATION_PREFIX) ||
+	    (chr == AT_PROP_NOTIFICATION_PREFX)) {
+		return true;
+	}
+
+	return false;
+}
+
+/**
+ * @brief Verify that the character is a valid character
+ *
+ * Notification ID strings can only contain upper case letters 'A' through 'Z'
+ *
+ * @param[in] char Character that should be examined
+ *
+ * @retval true  If character character is valid
+ * @retval false If character is not valid
+ */
+static inline bool is_valid_notification_char(char chr)
+{
+	chr = toupper(chr);
+
+	if ((chr >= 'A') && (chr <= 'Z')) {
+		return true;
+	}
+
+	return false;
+}
+
+/**
+ * @brief Check if the character identifies the end of the buffer
+ *
+ * All strings returned from the AT Host is null terminated, this function will
+ * check if we have hit the termination character.
+ *
+ * @param[in] char Character that should be examined
```

Comment:
Variable name is `chr`, not `char`. Please fix here and all below.

Reply:
Yes, all fixed